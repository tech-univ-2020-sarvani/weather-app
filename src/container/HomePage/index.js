import React from 'react';
import PropTypes from 'prop-types';
import details from '../../constants/weatherapp.json';
import useInput from '../../hooks/useInput';
import utils from '../../utils/useQuery';
// import weatherData from '../../constants/weatherData.json';

const HomePage = (props) => {
  const { testId } = props;
  const query = utils.useQuery();
  console.log(query);
  const latitude = query.get('lat');
  const longitude = query.get('lon');
  const cityname = query.get('cityname');
  let url = '';
  if (cityname) {
    url = `http://api.openweathermap.org/data/2.5/weather?q=${cityname}&appid=${details.key}`;
  } else {
    url = `http://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&appid=${details.key}`;
  }

  const [params, setParams, listLoadComplete] = useInput({}, url);
  if (listLoadComplete === null) {
    return (
      <div>Loading...</div>
    );
  }
  return (
    <div className="homepage" data-testId={testId}>
      <div className="card">
        <h3>
          Location:
          {' '}
          {params.name}
        </h3>
        <h3>
          latitude:
          {' '}
          {params.coord.lat}
        </h3>
        <h3>
          longitude:
          {params.coord.lon}
        </h3>
        <h3>
          weather:
          {params.weather[0].description}
        </h3>
        <h3>
          Min-temparature:
          {params.main.temp_min}
        </h3>
        <h3>
          Max-temparature:
          {params.main.temp_max}
        </h3>
      </div>
    </div>
  );
};

HomePage.propTypes = {
  testId: PropTypes.string,
};

export default HomePage;
