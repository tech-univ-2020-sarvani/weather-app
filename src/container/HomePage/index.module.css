.homepage{
    display: flex;
    justify-content: flex-start;
    flex-wrap: wrap;
}
.card{
    border: solid 2px black;
}