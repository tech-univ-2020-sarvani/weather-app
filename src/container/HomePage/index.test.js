import React from 'react';
import { render } from '@testing-library/react';
import HomePage from './index';
import utils from '../../utils/useQuery';

describe('The HomePage container', () => {
  it('should render correctly', () => {
    const mockuseQuery = jest.spyOn(utils, 'useQuery');
    mockuseQuery.mockResolvedValue({ get: jest.fn() });
    const { asFragment } = render(<HomePage testId="test-homepage" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should display Loading... when the api call is not complete', () => {

  });
});
