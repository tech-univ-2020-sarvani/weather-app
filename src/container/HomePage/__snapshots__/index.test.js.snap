// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The HomePage container should render correctly 1`] = `
<DocumentFragment>
  <div
    class="homepage"
    data-testid="test-homepage"
  >
    <button
      type="button"
    >
      Predict Weather
    </button>
    <input
      type="text"
    />
  </div>
</DocumentFragment>
`;
